templates/docs/examples/patterns/forms/_tristate_script.js

```diff
@@ -0,0 +1,31 @@
+/**
+  Sets up tri-state checkboxes and attaches event handlers to it
+  @param {String} selector The CSS selector matching tri-state checkboxes
+*/
+function initTristateCheckboxes(selector) {
+  var tristateCheckboxes = [].slice.call(document.querySelectorAll(selector));
+
+  tristateCheckboxes.forEach(function (checkbox) {
+    checkbox.addEventListener('change', function () {
+      var currentState = checkbox.getAttribute('aria-checked');
+
+      if (currentState === 'false') {
+        checkbox.setAttribute('aria-checked', 'mixed');
+        checkbox.checked = true;
+        checkbox.indeterminate = true;
+      } else if (currentState === 'mixed') {
+        checkbox.setAttribute('aria-checked', 'true');
```

Comment:
This is not needed. It's a native browser input, you don't need to set `aria-checked` on it, as setting `checked` does exactly the same.